v2/pkgs/tools.go

```diff
@@ -47,36 +49,37 @@ import (
 //	        └── 1.7.0
 //	            └── bossac
 //
-// It requires an Indexes client to list and read package index files: use the Indexes struct
+// It requires an Index Resource to search for tools
 type Tools struct {
-	Indexes interface {
-		List(context.Context) ([]string, error)
-		Get(context.Context, string) (Index, error)
-	}
+	Index  *index.Resource
 	Folder string
 }
 
 // Available crawles the downloaded package index files and returns a list of tools that can be installed.
 func (c *Tools) Available(ctx context.Context) (res tools.ToolCollection, err error) {
-	list, err := c.Indexes.List(ctx)
+	if !c.Index.IndexFile.Exist() || time.Since(c.Index.LastRefresh) > 1*time.Hour {
+		// Download the file again and save it
+		err := c.Index.DownloadAndVerify()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	body, err := os.ReadFile(c.Index.IndexFile.String())
```

Comment:
```suggestion
	body, err := t.Index.Read()
```
I'd avoid this code duplication and use a function that encapsulate such logic